File: backend-java/src/main/java/associados/api/mapper/AssociadoMapper.java
package associados.api.mapper;

import associados.api.dto.AssociadoForm;
import associados.api.model.Associado;
import org.springframework.stereotype.Component;

@Component
public class AssociadoMapper implements Mapper<Associado, AssociadoForm> {

    @Override
    public AssociadoForm map(Associado associado) {
        return new AssociadoForm(
                associado.getId(),
                associado.getNome(),
                associado.getObservacao(),
                associado.getCategoria(),
                associado.getDataNascimento(),
                associado.getNaturalidade(),
                associado.getEstadoNatural(),
                associado.getEstadoCivil(),
                associado.getNacionalidade(),
                associado.getEndereco(),
                associado.getBairro(),
                associado.getCidade(),
                associado.getEstado(),
                associado.getTelefone(),
                associado.getMesPago(),
                associado.getAnoPago()
        );
    }
}

File: backend/src/main/kotlin/associados/api/mapper/AssociadoMapper.kt
package associados.api.mapper

import associados.api.dto.AssociadoForm
import associados.api.model.Associado
import org.springframework.stereotype.Component

@Component
class AssociadoMapper: Mapper<AssociadoForm, Associado> {
    override fun map(t: AssociadoForm): Associado {
        return Associado(
            nome = t.nome,
            observacao = t.observacao,
            categoria = t.categoria,
            dataNascimento = t.dataNascimento,
            naturalidade = t.naturalidade,
            estadoNatural = t.estadoNatural,
            nacionalidade = t.nacionalidade,
            endereco = t.endereco,
            bairro = t.bairro,
            estado = t.estado,
            telefone = t.telefone,
            mesPago = t.mesPago,
            anoPago = t.anoPago,
            cidade = t.cidade,
            estadoCivil = t.estadoCivil
        )
    }
}